include/nanorange/algorithm/partial_sort_copy.hpp

```diff
@@ -7,23 +7,27 @@
 #ifndef NANORANGE_ALGORITHM_PARTIAL_SORT_COPY_HPP_INCLUDED
 #define NANORANGE_ALGORITHM_PARTIAL_SORT_COPY_HPP_INCLUDED
 
+#include <nanorange/algorithm/copy.hpp>
 #include <nanorange/algorithm/make_heap.hpp>
 #include <nanorange/algorithm/sort_heap.hpp>
 
 NANO_BEGIN_NAMESPACE
 
+template <typename I, typename O>
+using partial_sort_copy_result = copy_result<I, O>;
+
 namespace detail {
 
 struct partial_sort_copy_fn {
 private:
     template <typename I1, typename S1, typename I2, typename S2,
               typename Comp, typename Proj1, typename Proj2>
-    static constexpr I2 impl(I1 first, S1 last, I2 result_first,
+    static constexpr partial_sort_copy_result<I1, I2> impl(I1 first, S1 last, I2 result_first,
                              S2 result_last, Comp& comp, Proj1& proj1, Proj2& proj2)
     {
         I2 r = result_first;
         if (r == result_last) {
-            return r;
+            return {nano::next(first, last), std::move(result_first)};
```

Comment:
The change from

    return {std::move(nano::next(first, last), std::move(result_first)};

to

    return {nano::next(first, last), std::move(result_first)};

causes gcc 9 to ICE in the BUILD_SINGLE_HEADER job. Should I add `std::move` back?

Reply:
Well, that's very strange! Yes, please put the `std::move` back to avoid the ICE, and add a brief comment